File: aoc-2015/src/test/java/com/capital7software/aoc2015aoc/days/Day24Test.java
package com.capital7software.aoc2015aoc.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.aoc2015aoc.days.Day24;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day24Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day24Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day24();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getLowestQeScoreWith3Partitions() {
    var instance = new Day24();
    int expected = 99;
    var actual = instance.getLowestQeScore(lines, 3);
    assertEquals(expected, actual, "The lowest QE Score with 3 partitions "
        + "is not what was expected: " + expected);
  }

  @Test
  void getLowestQeScoreWith4Partitions() {
    var instance = new Day24();
    int expected = 44;
    var actual = instance.getLowestQeScore(lines, 4);
    assertEquals(expected, actual, "The lowest QE Score with 4 partitions "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day24Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day24Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day24Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day24()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testGetStrongestBridge() {
    val instance = Day24()
    val expected = 31
    val actual = instance.getStrongestBridge(lines).sumOf { it.strength }

    Assertions.assertEquals(
        expected, actual, "$actual for the strongest bridge strength " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testGetLongestThenStrongestBridge() {
    val instance = Day24()
    val expected = 19
    val actual = instance.getLongestThenStrongestBridge(lines).sumOf { it.strength }

    Assertions.assertEquals(
        expected, actual, "$actual for the longest then strongest bridge strength " +
        "is not what was expected: $expected"
    )
  }
}
